networkx/algorithms/perfect_graph.py

```diff
@@ -0,0 +1,37 @@
+import networkx as nx
+
+__all__ = ["is_perfect_graph"]
+
+
+def is_perfect_graph(G):
+    """Return True if G is a perfect graph, else False.
+
+    According to the Strong Perfect Graph Theorem, a graph is perfect
+    if and only if neither it nor its complement contains an induced
+    odd cycle of length at least 5.
+    """
+
+    def has_induced_odd_hole(graph):
+        nodes = list(graph.nodes)
+        n = len(nodes)
+        for i in range(n):
+            for j in range(i + 1, n):
+                try:
+                    paths = nx.all_simple_paths(graph, nodes[i], nodes[j], cutoff=n)
+                except nx.NetworkXNoPath:
+                    continue
+                for path in paths:
+                    if len(path) < 5 or len(path) % 2 == 0:
+                        continue
+                    if graph.has_edge(path[-1], path[0]):
+                        cycle_nodes = path
+                        subG = graph.subgraph(cycle_nodes)
+                        if subG.number_of_edges() == len(cycle_nodes):
+                            return True
+        return False
```

Comment:
IMO this would be better as an external "private" function (i.e. `def _has_induced_odd_hole`) rather than a nested function. Doing so would have several advantages:
 1. Users don't have to worry about multiple scopes when reading the code (though arguably that's trivial in this case as there are no variables that cross scope boundaries)
 2. It allows the function to be independently tested
 3. It makes it easier to replace the function in the future if a better implementation is found.